emukit/model_wrappers/gpy_quadrature_wrappers.py

```diff
@@ -162,26 +164,57 @@ def dKdiag_dx(self, x: np.ndarray) -> np.ndarray:
-        lower and upper bounds of the integral i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+    lower and upper bounds of the integral i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]. None means infinite
+    bounds.
+    :param measure: an integration measure. None means no measure.
     :param integral_name: the (variable) name(s) of the integral
 
-    :return: emukit model for quadrature witg GPy backend (IBaseGaussianProcessGPy)
+    :return: emukit model for quadrature with GPy backend (IBaseGaussianProcessGPy)
     """
 
-    # wrap standard kernel and get quadrature kernel
-    if gpy_model.kern.name is 'rbf':
+    # wrap standard kernel
+    if gpy_model.kern.name == 'rbf':
         standard_kernel_emukit = RBFGPy(gpy_model.kern)
-        quadrature_kernel_emukit = QuadratureRBF(standard_kernel_emukit, integral_bounds=integral_bounds,
-                                                 integral_name=integral_name)
     else:
-        raise NotImplementedError("Only the GPy rbf-kernel is supported right now.")
+        raise ValueError('Only RBF kernel is supported. Got ', gpy_model.kern.name, ' instead.')
+
+    # check if measure and bounds fit together and wrap combination in a quadrature kernel
+    # infinite bounds and no measure. Can't do.
+    if (integral_bounds is None) and (measure is None):
+        raise ValueError('integral_bounds are infinite and measure is None. Choose either finite bounds or an '
+                         'appropriate integration measure.')
+
+    # infinite bounds. Gauss yes, uniform measure not implemented yet
+    elif (integral_bounds is None) and (measure is not None):
+        if isinstance(measure, UniformMeasure):
+            raise ValueError('Q-kernel for uniform measure not implemented yet.')
+        elif isinstance(measure, IsotropicGaussianMeasure):
+            quadrature_kernel_emukit = QuadratureRBFIsoGaussMeasure(rbf_kernel=standard_kernel_emukit,
+                                                                    measure=measure,
+                                                                    variable_names=integral_name)
+        else:
+            raise ValueError('Currently only IsotropicGaussianMeasure supported with these integral bounds.')
+
+    # finite bounds, no measure
+    elif (integral_bounds is not None) and (measure is None):
+        quadrature_kernel_emukit = QuadratureRBFnoMeasure(rbf_kernel=standard_kernel_emukit,
+                                                          integral_bounds=integral_bounds,
+                                                          variable_names=integral_name)
+
+    # finite bounds: measure: uniform only not implemented
+    else:
+        if isinstance(measure, UniformMeasure):
+            raise ValueError('Q-kernel for uniform measure and finite bounds not implemented yet.')
+        else:
+            raise ValueError('Currently only None measure supported with these integral bounds.')
```

Comment:
I must be missing something obvious here. How can an exception be avoided? All branches seem to lead to one

Also, this last message isn't clear. What is `None` measure? And it would be better to name things explicitly intead of referring to `these bounds`

Reply:
point 1: yes, the first error is a placeholder for the kernel which will come after this PR. I had not included it because the PR was already big. 

point 2: I replaced it with "Currently only measure=None is supported with these integral bounds."